Assets/Scripts/Mixed/Systems/KinematicCharacterControllerSystems.cs

```diff
@@ -0,0 +1,189 @@
+
+using PropHunt.Mixed.Components;
+using PropHunt.Mixed.Utilities;
+using Unity.Burst;
+using Unity.Entities;
+using Unity.Mathematics;
+using Unity.NetCode;
+using Unity.Physics;
+using Unity.Physics.Systems;
+using Unity.Transforms;
+
+namespace PropHunt.Mixed.Systems
+{
+    /// <summary>
+    /// Updates the grounded data on a kinematic character controller
+    /// </summary>
+    [BurstCompile]
```

Comment:
I haven't seen any docs regarding using the **[BurstCompile]** Attribute on the ComponentSystems.
I don't know if this actually work (If you're sure it does, then just ignore my comment).

The way I'd usually do this is by declaring a SystemBase instead of a ComponentSystem, and then do an .ScheduleParallel() on the foreach.

i.e
```csharp
Entities.Foreach((.. references and components) => {
...
...
}).ScheduleParallel();
``` 

This will allow you to jobify your System Update methods, and automatically apply BurstCompilation (Without the need of the attribute).

The reason why I'm bringing this up is because I thought only Jobified code could be Burst compiled, and AFAIK a ComponentSystem by itself it's not automatically jobified.

Reply:
I'll make a separate issue for handling this (god knows this one is too long 😆) . I believe the [Burst](https://docs.unity3d.com/Packages/com.unity.burst@1.3/manual/index.html) attribute just changes how the code is compiled and executed. It runs it utilizing some naturally parallelized format and using gpu/multi core capabilities so I'm not certain how it would work. 

Additionally, there is one large problem with the character controller, each movement can be projected for MaxBounces number of moves. Meaning it may need to do somewhere between 2-5 raycast jobs depending on how much it bounces. This leads to lots of idle time when processing the job in parallel so the [Project Valid Movement](https://github.com/nicholas-maltbie/PropHunt/blob/56dba8ac5021b390b4d651ebbda5fafadd2ab929/Assets/Scripts/Mixed/Utility/KinematicCharacterControllerUtilities.cs#L55) function would have to be rethought with this batching and run as a recursive, stateless function as opposed to the current layout for better use.

The KCC Grounded, Jump, and Gravity from [KCCSystems](https://github.com/nicholas-maltbie/PropHunt/blob/feature/KinematicCharacter/Assets/Scripts/Mixed/Systems/KinematicCharacterControllerSystems.cs) could all be jobified much more easily and telling them to be explicitly parallel would improve performance. I will break those into a separate issue.